source/opt/const_folding_rules.cpp

```diff
@@ -196,7 +254,84 @@ double GetDoubleFromConst(const analysis::Constant* c) {
   }
 }
 
-// This macro defines a |FloatScalarFoldingRule| that applies |op|.  The
+// Returns the integer value of |c|.  The constant |c| must have type
+// |Integer|, and width |32|.
+uint32_t GetIntegerFromConst(const analysis::Constant* c) {
```

Comment:
Alan add some functions to the Constant class to do this.  When you rebase you should be able to use them.

Reply:
OK